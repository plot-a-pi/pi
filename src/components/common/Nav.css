body{
    background-color: #570963;
}

.NavBackground{
    animation: changeBackground 1.5s 1 forwards, slideInToTheRight 1s 1 forwards;
    width: 20vw;
}

.Nav{
    display: flex;
    justify-content: start;
    flex-direction: column;
    width: 20vw;
    height: 100vh;
    transition-timing-function: ease-in;
    font-size: 2.5vw;
    font-family: Impact, Haettenschweiler, 'Arial Narrow Bold', sans-serif;
    color:black;
    animation: slideInToTheRight 1.5s 1 forwards;
        input{
            visibility: none;
        }
}

.Collumn{
    display: flex;
    justify-content: start;
    flex-direction: column;
    transition: ease-in;
    animation: slowRotate 1.5s alternate forwards;
}
.Big{
    font-size: 10vh;
    max-width: 10vw;
    margin-left: 0px;
    transform: rotate(90deg);
    transform: skewY(10vh);
}
@keyframes slowRotate{
    0%{
        transform: (0.586,0.8,-0.8,0.586,40,20);
    }
    20%{
        transform: matrix(0.586,0.8,-0.8,0.586,40,20);
    }
    60%{
        transform: matrix(0.586,0.8,-0.8,0.586,40,20);
    }
    100%{
        font-size: 10vh;
    }
}
@keyframes slideInToTheRight {
    0%{
        transform: translateX(-50vw);
    }
    20%{
        transform: translateX(-20vw);
    }
    60%{
        transform: translateX(-5vw);
    }
    100%{
        transform: translateX(0vw);
    }
}
@keyframes changeBackground {
    0%{
        background-color: #570963;
    }
    100%{
        background-color: rgba(221, 157, 231, 1);
    }
}