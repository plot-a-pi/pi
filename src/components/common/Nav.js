import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { useNav } from '../../hooks/nav';
import { signOut } from '../../firebase/firebase';
import styles from './Nav.css';

const Nav = () => {
  const { handleChange } = useNav();
  const [publicFacingNav, setPublicFacingNav] = useState(false);

  const handleClick = () => {
    signOut();
  };

  let jsx;
  if(!publicFacingNav){
    jsx = (
      <div className={styles.Big}>
        <span style={{ color :'rgba(221, 157, 231, 1)' }} onClick={() => setPublicFacingNav(!publicFacingNav)}>|||</span>
      </div>);
  }
  if(publicFacingNav){
    jsx = (
      <div className={styles.NavBackground}>
        <div id='jsxHorizontal' className={styles.Column}>
          <span style={{ color: '#570963', textDecoration: 'none' }} onClick={() => setPublicFacingNav(!publicFacingNav)}>|||</span>
        </div>
        <button onClick={handleClick}>Sign Out</button>
        <ul className={styles.Nav}>
          <NavLink style={{ color: 'blue', textDecoration: 'none' }} exact to='/' activeStyle={{
            fontWeight: 'bold',
            backgroundColor: 'purple'
          }}>
            <input style={{ display: 'none' }} id='home' value='home' onChange={handleChange} />
            <label htmlFor='home'>Home</label>
          </NavLink>
          <br/>
          <NavLink style={{ color: 'blue', textDecoration: 'none' }}  to='/teachers' activeStyle={{
            fontWeight: 'bold',
            backgroundColor: 'purple'
          }}>
            <input style={{ display: 'none' }} id='teachers' value='teachers'  onChange={handleChange} />
            <label htmlFor='teachers'>For Groups</label>
          </NavLink>
          <br/>
          <NavLink style={{ color: 'blue', textDecoration: 'none' }}  to='/pifaqs' activeStyle={{
            fontWeight: 'bold',
            backgroundColor: 'purple'
          }}>
            <input style={{ display: 'none' }} id='pifaqs' value='pifaqs'  onChange={handleChange} />
            <label htmlFor='pifaqs'>Pi FAQs</label>
          </NavLink>
          <br/>
          <NavLink style={{ color: 'blue', textDecoration: 'none' }}  to='/map' activeStyle={{
            fontWeight: 'bold',
            backgroundColor: 'purple'
          }}>
            <input style={{ display: 'none' }} id='map' value='map'  onChange={handleChange} />
            <label htmlFor='map'>Data Map</label>
          </NavLink>
          <br/>
          <NavLink style={{ color: 'blue', textDecoration: 'none' }}  to='/montecarlo' activeStyle={{
            fontWeight: 'bold',
            backgroundColor: 'purple'
          }}>
            <input style={{ display: 'none' }} id='monteCarlo' value='monteCarlo'  onChange={handleChange} />
            <label htmlFor='monteCarlo'>Monte Carlo</label>
          </NavLink>
        </ul>
      </div>);
  }
  return (jsx);
};

export default Nav;

