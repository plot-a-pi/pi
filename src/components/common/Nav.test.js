import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';

describe('Nav component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper).toMatchSnapshot();
  });
});
