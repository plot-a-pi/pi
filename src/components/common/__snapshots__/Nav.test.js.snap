// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches a snapshot 1`] = `
<div
  className="Big"
>
  <span
    onClick={[Function]}
    style={
      Object {
        "color": "rgba(221, 157, 231, 1)",
      }
    }
  >
    |
  </span>
  <span
    onClick={[Function]}
    style={
      Object {
        "color": "rgba(221, 157, 231, 1)",
      }
    }
  >
    |
  </span>
  <span
    onClick={[Function]}
    style={
      Object {
        "color": "rgba(221, 157, 231, 1)",
      }
    }
  >
    |
  </span>
</div>
`;
